src/visibility.js

```diff
@@ -192,24 +200,35 @@ class Visibility extends BasePlugin {
     this.eventManager.destroy();
   }
 
-  _dragMouseDown(e: MouseEvent) {
-    e.preventDefault();
+  _startDrag(e: any, moveEventName: string, endEventName: string) {
+    this.eventManager.listenOnce(document, endEventName, () => {
+      this._stopDrag();
+    });
+
     // get the mouse cursor position at startup:
-    this._currMousePos.x = e.clientX;
-    this._currMousePos.y = e.clientY;
-    this.eventManager.listen(document, 'mouseup', this._dragStop.bind(this));
-    this.eventManager.listen(document, 'mousemove', this._floatingDrag.bind(this));
+    this._currMousePos.x = this._clientX(e);
+    this._currMousePos.y = this._clientY(e);
+
+    this.eventManager.listen(document, moveEventName, () => {
+      this._moveDrag();
+    });
+  }
+  _clientX(e: any): number {
+    return e.clientX || (e.changedTouches && e.changedTouches[0] && e.changedTouches[0].clientX);
+  }
+  _clientY(e: any): number {
+    return e.clientY || (e.changedTouches && e.changedTouches[0] && e.changedTouches[0].clientY);
   }
 
-  _floatingDrag(e: MouseEvent) {
+  _moveDrag(e: any) {
```

Comment:
can we make this type stricter? MouseEvent | Event?

Reply:
In some i changed (pushed it now) and some i couldn't